External/SPMmodified/xASL/xASL_im_FillNaNs.m

```diff
@@ -49,10 +49,9 @@
 % Copyright (C) 2015-2024 ExploreASL
 
 %% Admin
-IM = NaN; % default
-
 if nargin<1 || isempty(InputPath)
     warning('Illegal input path, skipping');
+	IMout = NaN;
```

Comment:
You removed the default of IM = NaN, and moved it to the clause of nargin<1 only. So if there are any other errors in the code, this script not only gives a warning but also crashes because IMout is not assigned.

That was the reason to default it to NaN before anything else. Any (unforeseen) warning or issue that might occur, will issue a warning and the code will continue.

Reply:
All unforseen errors give the correct output with my solution. Functionality is perfectly fine... The reason was that it wouldn't work with the dimension swap with an initialized IMout.

But if you really insist, I can change it back and think of another solution.

Reply:
OK let's keep it like this then